packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -394,24 +394,42 @@ class Border extends BoxBorder {
 
   @override
   bool get isUniform {
+    if (!_colorIsUniform)
+      return false;
+
+    if (!_widthIsUniform)
+      return false;
+
+    if (!_styleIsUniform)
+      return false;
+
+    return true;
+  }
+
+  bool get _colorIsUniform {
     final Color topColor = top.color;
     if (right.color != topColor ||
-        bottom.color != topColor ||
-        left.color != topColor)
+      bottom.color != topColor ||
+      left.color != topColor)
       return false;
+    return true;
+  }
 
+  bool get _widthIsUniform {
     final double topWidth = top.width;
     if (right.width != topWidth ||
-        bottom.width != topWidth ||
-        left.width != topWidth)
+      bottom.width != topWidth ||
+      left.width != topWidth)
       return false;
+    return true;
```

Comment:
Can this (and the other two above and below) be written in a single return statement:

```suggestion
    return right.width == topWidth && bottom.width == topWidth && left.width == topWidth;
```

(Please double check that I transformed the logic correctly...